alphagov/forms-admin
# 315-ask-answer-type-first-when-creating-question-in-admin

#### What problem does the pull request solve?

Trello card: https://trello.com/c/N78PUM5K/315-ask-answer-type-first-when-creating-question-in-admin

This PR introduces a new flow for creating a question for a form. This includes a page where the user selects the answer type for their new question, and then sets other question details. To do this, the answer type is stored in session, which is new for this application. 

**Add a question**

To test, add a new question to a form:

![image](https://user-images.githubusercontent.com/25597009/204855730-bbbb93e5-b778-4dce-8043-416135979fd2.png)

You should then be able to select an answer type:

![image](https://user-images.githubusercontent.com/25597009/204855849-7af4ee86-df97-46bc-8886-2d1e419ba85a.png)

And on submission, you should be able to review the answer type that's been selected:

![image](https://user-images.githubusercontent.com/25597009/204856011-1f84be9b-08c8-4569-a59f-6e2c0fc05132.png)

**Edit an existing question**

Click on edit for an existing question:

![image](https://user-images.githubusercontent.com/25597009/204856417-da54ae90-52cb-425a-bdc2-e400be024a5b.png)

You are able to see the answer type in the summary, with a change link:

![image](https://user-images.githubusercontent.com/25597009/204856544-f49d71c9-8f64-49fa-a11e-ad397b9c48d1.png)

Clicking on the change link will bring you to the pre-populated answer type list:

![image](https://user-images.githubusercontent.com/25597009/204856641-b05c7dd0-eab6-4643-b498-412c4457b6ed.png)

Selecting a new answer type will save this to the API, and return to the Edit page.



#### Checklist

- [ ] I've used the pull request template
- [ ] I've linked this PR to the relevant issue (if mission work)
- [ ] I've written unit tests for these changes (if code change)
- [ ] I've updated the documentation in (If any documentation requires updating)
  - [ ] Elsewhere (please link)
